File: common/src/main/java/com.codeabovelab.tpc/doc/DocumentField.java
package com.codeabovelab.tpc.doc;

import com.codeabovelab.tpc.text.Textual;

/**
 */
public interface DocumentField extends Textual {

    interface Builder {
        DocumentFieldImpl build(Document.Builder document);
    }

    /**
     * Filed name. Note that it not does not include document id.
     * @return non null string
     */
    String getName();
}

File: common/src/main/java/com.codeabovelab.tpc/doc/DocumentField.kt
package com.codeabovelab.tpc.doc

import com.codeabovelab.tpc.text.Textual

/**
 */
interface DocumentField : Textual {

    interface Builder<B : Builder<B>>: Textual.Builder<B> {
        override fun build(): DocumentField
    }
}
